ksqldb-streams/src/test/java/io/confluent/ksql/execution/streams/SourceBuilderTest.java

```diff
@@ -527,6 +531,20 @@ public void shouldAddRowTimeAndRowKeyColumnsToNonWindowedStream() {
     assertThat(withTimestamp, equalTo(GenericRow.genericRow("baz", 123, A_ROWTIME, A_KEY)));
   }
 
+  @Test
+  public void shouldAddRowPartitionAndOffsetColumnsToNonWindowedStream() {
+    // Given:
+    givenUnwindowedSourceStream(SystemColumns.ROWPARTITION_ROWOFFSET_PSEUDOCOLUMN_VERSION);
+    final ValueTransformerWithKey<GenericKey, GenericRow, GenericRow> transformer =
+        getTransformerFromStreamSource(streamSource);
+
+    // When:
+    final GenericRow withTimestamp = transformer.transform(KEY, row);
+
+    // Then:
+    assertThat(withTimestamp, equalTo(GenericRow.genericRow("baz", 123, A_ROWTIME, A_ROWPARTITION, A_ROWOFFSET, A_KEY)));
+  }
+
```

Comment:
Should I add new tests like this? Basically every test in the file is checking the value columns of rows, so it's a debate between changing `shouldAddRowTimeAndRowKeyColumnsToNonWindowedStream()` to this and making the logic more discrete. I lean towards the former as I don't see a huge advantage to the latter outside of testing for the newer feature explicitly (but it shouldn't be horribly difficult to spot any bugs related directly to these fields failing anyway)

Reply:
We should have two versions of this test: one for `ROWTIME_PSEUDOCOLUMN_VERSION` and one for `ROWPARTITION_ROWOFFSET_PSEUDOCOLUMN_VERSION`, since the expected behavior is different and both are important. In other words, I think it's good that you added these two new tests. Be aware that once `CURRENT_PSEUDOCOLUMN_VERSION_NUMBER` is bumped to 1, the existing versions of these tests are going to fail. At that time they'll be redundant with the new test that explicitly tests `ROWPARTITION_ROWOFFSET_PSEUDOCOLUMN_VERSION`, so if you want you can update the existing version to explicitly use `ROWTIME_PSEUDOCOLUMN_VERSION` now, in order to save yourself some trouble later. (You won't be saving too much trouble later though since the rest of the test that use `CURRENT_PSEUDOCOLUMN_VERSION_NUMBER` will also need to be updated.)

We should also add new versions of the tests `shouldAddRowTimeAndTimeWindowedRowKeyColumnsToStream()` and `shouldAddRowTimeAndTimeWindowedRowKeyColumnsToTable()`.

Reply:
Sounds good. I think it makes sense to me to leave the tests which will become outdated all in a position that they'll fail when `CURRENT_PSEUDOCOLUMN_VERSION_NUMBER` is changed.